application/heartbeater.go

```diff
@@ -70,3 +82,9 @@ func (hb *heartbeater) send(tags map[string]string) {
 		log.Printf("heartbeater SendMetric error: %v\n", err)
 	}
 }
+
+func (hb *heartbeater) AddCustomTags(tags map[string]string) {
+	hb.mux.Lock()
+	hb.customTags = append(hb.customTags, tags)
```

Comment:
Since we do not have a Set, and are using a slice instead, we need to verify we are not appending the same map[k]->values that already exist in our list of customTags.

We should not simply append the tags. Only append if the input tags map is different than what we already have amongst `hb.customTags`.